scripts/merge_truth_per_tract.py

```diff
@@ -162,8 +163,15 @@ def match_object_with_merged_truth(
     # Calculate magnitude difference for all the pairs
     with np.errstate(invalid="ignore"):
         dmag = np.abs(
-            object_cat.loc[object_idx, mag_label_obj].values -
-            truth_cat.loc[truth_idx, mag_label_truth].values
+            np.where(
+                (
+                    (truth_cat.loc[truth_idx, "truth_type"].values == 1)
+                    & np.isfinite(object_cat.loc[object_idx, mag_label_obj_cm].values)
+                ),
+                object_cat.loc[object_idx, mag_label_obj_cm].values,
```

Comment:
Add a comment explaining what this selection is doing here.  Seeing the `_cm` vs `_ps` can be a little hard in all of the other code.  "Match based on cModel magnitudes for true galaxy with finite measured cModel magnitudes.  Otherwise match on PSF magnitude."